main/screen.c

```diff
@@ -403,6 +436,45 @@ static void screen_update_cb(lv_timer_t * timer)
         lv_label_set_text_fmt(chip_temp_label, "Temp: %.1f C", power_management->chip_temp_avg);
     }
 
+    if (wifi_rssi_value_label) {
+        char rssi_buf[25];
+        
+        if (module->wifi_rssi < 0 && module->wifi_rssi >= -127) { // Typical RSSI range
+            snprintf(rssi_buf, sizeof(rssi_buf), "RSSI: %d dBm", module->wifi_rssi);
+        } else {
+            snprintf(rssi_buf, sizeof(rssi_buf), "RSSI: -- dBm");
+        }
+        if (strcmp(lv_label_get_text(wifi_rssi_value_label), rssi_buf) != 0) {
+            lv_label_set_text(wifi_rssi_value_label, rssi_buf);
+        }
+    }
+
+    if (esp_uptime_label) {
+        char uptime[50];
```

Comment:
This whole thing gets called on every `screen_update_cb`, which is a lot. It might make sense to do this in a timer that runs once every second and set a char * field in SYSTEM_MODULE.